build.gradle

```diff
@@ -760,6 +763,7 @@ project(':iceberg-hive-metastore') {
     testImplementation project(path: ':iceberg-api', configuration: 'testArtifacts')
     testImplementation project(path: ':iceberg-core', configuration: 'testArtifacts')
     testImplementation libs.awaitility
+    testImplementation 'com.zaxxer:HikariCP:5.1.0'
```

Comment:
Spark 4.0 removed dependency on `com.jolbox:bonecp`. In `TestHiveMetastore.java`, I explicitly set the connection pool to `conf.set("datanucleus.connectionPoolingType", "HikariCP");`, otherwise, it will default to ``com.jolbox:bonecp` and we will get 
```
Exception in thread "TThreadPoolServer WorkerProcess-5" java.lang.NoClassDefFoundError: com/jolbox/bonecp/BoneCPConfig
	at org.apache.hadoop.hive.metastore.txn.TxnHandler.setupJdbcConnectionPool(TxnHandler.java:3156)
	at org.apache.hadoop.hive.metastore.txn.TxnHandler.setConf(TxnHandler.java:260)
```
Since I explicitly set the connection pool to `conf.set("datanucleus.connectionPoolingType", "HikariCP");`, we need to add the test dependency.

Reply:
DBCP(already pulled by Spark, see SPARK-48538) can be used to avoid pulling HikariCP dependency

Reply:
somehow after I changed to `DBCP`, I started to have 
```
> Task :iceberg-spark:iceberg-spark-runtime-4.0_2.13:integrationTest FAILED
```
It doesn't have any additional information. 

If I run  `./gradlew :iceberg-spark:iceberg-spark-runtime-4.0_2.13:integrationTest` on my local, it always passes.

Now even after I switched back to `HikariCP`. `Task :iceberg-spark:iceberg-spark-runtime-4.0_2.13:integrationTest ` still failed.

Reply:
should be irrelevant, could you try uploading the reports for the failure test?

`.github/workflows/spark-ci.yml`
```patch
      - uses: actions/upload-artifact@v4
        if: failure()
        with:
          name: test logs
          path: |
            **/build/testlogs
+           **/build/reports
```

Reply:
what's the root cause?

Reply:
```
Caused by: org.junit.platform.commons.JUnitException: OutputDirectoryProvider not available; probably due to unaligned versions of the junit-platform-engine and junit-platform-launcher jars on the classpath/module path.
```
I guess `junit-platform-engine` has been upgraded in upstream. I rebased and the problem went away.